azure/types.go

```diff
@@ -409,6 +409,9 @@ type AgentPoolSpec struct {
 	// Minimum number of nodes for auto-scaling
 	MinCount *int32 `json:"minCount,omitempty"`
 
+	// Node labels - labels for all of the nodes present in node pool
+	NodeLabels map[string]*string `json:"nodeLabels,omitempty"`
```

Comment:
What's the value of the `*string` value here? Do we really want to permit a key string with a nil value?

Reply:
I will change it. The initial motivation was to avoid conversion from `map[string]string` to `map[string]*string`

Reply:
done

Reply:
wait, not done but I changed it in `azuremanagedmachinepool_types.go` as in our API types yes I agree that there is no reason to use `*string`
but ehere I kept the same as the one expected in `ManagedClusterAgentPoolProfileProperties`